hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -895,9 +895,17 @@ private void insertCustomFees(List<com.hederahashgraph.api.proto.java.CustomFee>
         }
     }
 
-    private void parseFixedFee(CustomFee customFee, FixedFee fixedFee) {
+    private void parseFixedFee(CustomFee customFee, FixedFee fixedFee, EntityId tokenId) {
         customFee.setAmount(fixedFee.getAmount());
-        customFee.setDenominatingTokenId(EntityId.of(fixedFee.getDenominatingTokenId()));
+
+        if (fixedFee.hasDenominatingTokenId()) {
+            TokenID denominatingTokenId = fixedFee.getDenominatingTokenId();
+            if (denominatingTokenId.equals(TokenID.getDefaultInstance())) {
+                customFee.setDenominatingTokenId(tokenId);
+            } else {
+                customFee.setDenominatingTokenId(EntityId.of(denominatingTokenId));
+            }
```

Comment:
nit: Can avoid the duplication of setter:
```suggestion
            EntityId denominatingTokenId = EntityId.of(fixedFee.getDenominatingTokenId());
            customFee.setDenominatingTokenId(denominatingTokenId == EntityId.EMPTY ? tokenId : denominatingTokenId);
```